tuf/ngclient/_internal/trusted_metadata_set.py

```diff
@@ -226,6 +226,10 @@ def update_timestamp(self, data: bytes) -> Metadata[Timestamp]:
                     f"New timestamp version {new_timestamp.signed.version} must"
                     f" be >= {self.timestamp.signed.version}"
                 )
+            # If new_timestamp.version == trusted_timestamp.version ignore it.
+            if new_timestamp.signed.version == self.timestamp.signed.version:
+                return self.timestamp
```

Comment:
If we keep using the old timestamp, we should check its expiry (see self._check_final_timestamp()). cc @jku

Reply:
this happens in the beginning of update_snapshot() -- when we know we should have final timestamp.

Reply:
I see, and we check the new one here too, so that we don't persist an already expired one?

Reply:
The timestamp case doesn't actually need the earlier check or anything but it does this for consistency with other methods: it's done so that Updater can 
* successfully load the local metadata for checks even if metadata is not valid as final but also
* avoid fetching the remote metadata completely if local metadata is valid as final

both cases are relevant for snapshot and targets, but we don't actually need the second case for timestamp.

Reply:
The check does save us a write operation, in case remote metadata is already expired, right? IIUC, without the check we'd persist remote metadata regardless of expiration.